### ISSUE TITLE
Out of range issue

### ISSUE THREAD

**Issue by arthurwolf** [c9s/bbgo#386]
```
        /home/arthur/go/pkg/mod/github.com/c9s/bbgo@v1.20.1-0.20211209133023-920462b6f752/pkg/backtest/stream.go:79 +0x6ba
arthur@crypto:~/bbgo$ timeout --kill-after=630s 600s /home/arthur/bbgo/bbgo backtest --exchange binance --dotenv /tmp/bbgo-env/bbgo-env-KAVAUSDT.yaml --config /tmp/bbgo-config/37236810dc639fd9269d97f98ae45bc3.yaml --base-asset-baseline --output /tmp/bbgo-cache/37236810dc639fd9269d97f98ae45bc3/ 2>&1
[0000]  INFO starting backtest with startTime Wed Jan  8 00:00:00 2020
[0001]  INFO querying balances from session binance... session=binance
[0001]  INFO binance account session=binance
[0001]  INFO  USDT: 10000.000000
panic: runtime error: index out of range [4901] with length 4901

goroutine 84 [running]:
github.com/c9s/bbgo/pkg/indicator.(*EWMA).calculateAndUpdate(0xc0006acd80, {0xc001be1480, 0x1327, 0x17ba})
        /home/arthur/go/pkg/mod/github.com/c9s/bbgo@v1.20.1-0.20211209133023-920462b6f752/pkg/indicator/ewma.go:82 +0x818
github.com/c9s/bbgo/pkg/indicator.(*EWMA).handleKLineWindowUpdate(...)
        /home/arthur/go/pkg/mod/github.com/c9s/bbgo@v1.20.1-0.20211209133023-920462b6f752/pkg/indicator/ewma.go:146
github.com/c9s/bbgo/pkg/bbgo.(*MarketDataStore).EmitKLineWindowUpdate(0x1050f60, {0xc000ac8288, 0x2}, {0xc001be1480, 0x1327, 0x17ba})
        /home/arthur/go/pkg/mod/github.com/c9s/bbgo@v1.20.1-0.20211209133023-920462b6f752/pkg/bbgo/marketdatastore_callbacks.go:15 +0x65
github.com/c9s/bbgo/pkg/bbgo.(*MarketDataStore).AddKLine(0xc0006a2ba0, {0x31c94, {0xc000ac8280, 0x7}, {0xc000ac8290, 0x8}, {0x0, 0xed5a8f2f8, 0x0}, {0x0, ...}, ...})
        /home/arthur/go/pkg/mod/github.com/c9s/bbgo@v1.20.1-0.20211209133023-920462b6f752/pkg/bbgo/marketdatastore.go:62 +0x208
github.com/c9s/bbgo/pkg/bbgo.(*MarketDataStore).handleKLineClosed(...)
        /home/arthur/go/pkg/mod/github.com/c9s/bbgo@v1.20.1-0.20211209133023-920462b6f752/pkg/bbgo/marketdatastore.go:47
github.com/c9s/bbgo/pkg/types.(*StandardStream).EmitKLineClosed(0xc0009f9d40, {0x31c94, {0xc000ac8280, 0x7}, {0xc000ac8290, 0x8}, {0x0, 0xed5a8f2f8, 0x0}, {0x0, ...}, ...})
        /home/arthur/go/pkg/mod/github.com/c9s/bbgo@v1.20.1-0.20211209133023-920462b6f752/pkg/types/standardstream_callbacks.go:83 +0x7a
github.com/c9s/bbgo/pkg/backtest.(*Stream).Connect.func2()
        /home/arthur/go/pkg/mod/github.com/c9s/bbgo@v1.20.1-0.20211209133023-920462b6f752/pkg/backtest/stream.go:96 +0x2a5
created by github.com/c9s/bbgo/pkg/backtest.(*Stream).Connect
        /home/arthur/go/pkg/mod/github.com/c9s/bbgo@v1.20.1-0.20211209133023-920462b6f752/pkg/backtest/stream.go:79 +0x6ba
arthur@crypto:~/bbgo$ cat /tmp/bbgo-config/37236810dc639fd9269d97f98ae45bc3.yaml
---
sessions:
  binance:
    exchange: binance
    envVarPrefix: binance
backtest:
  startTime: "2020-01-08"
  endTime: "2020-01-15"
  symbols:
  - KAVAUSDT
  account:
    makerCommission: 15 # TODO: Set the right values for binance, figure out what the number means
    takerCommission: 15
    balances:
      KAVA: 0.0
      USDT: 10000
exchangeStrategies:
- on: binance
  grid:
    symbol: KAVAUSDT
    # was: quantity: 1428.57142857142866887443
    amount: 1428.57142857142866887443
    gridNumber: 7.00000000000000000000
    profitSpread: 0.03319934615652114784
    upperPrice: 1.12768088037109359689
    lowerPrice: 1.01294509765624995978
    long: true
arthur@crypto:~/bbgo$ ./bbgo -v
```

**Comment by arthurwolf** [comment]
After upgrading to the latest commit, a new error appears:

```
packagefile google.golang.org/protobuf/internal/descfmt=/home/arthur/.cache/go-build/de/deff37091a517c0b9184605964f6fcd2042c5abe1a053ffec9f5b0b83cf5b342-d
packagefile google.golang.org/protobuf/internal/descopts=/home/arthur/.cache/go-build/ff/ff3edbfd9066e1838301351b367009c0d92f9d98187ae1d049b29a64e3f2d7bb-d
packagefile google.golang.org/protobuf/internal/encoding/defval=/home/arthur/.cache/go-build/45/45919a06c0f56a2d88d05bc1bcf3e7f9c6b2740104e9a68b04350950aab36248-d
packagefile google.golang.org/protobuf/internal/encoding/tag=/home/arthur/.cache/go-build/75/759610c924fd58539830eb68edc93776f1cf2f17931e34ee91fe055f94e35fb9-d
EOF
mkdir -p $WORK/b001/exe/
cd .
/usr/local/go/pkg/tool/linux_amd64/link -o $WORK/b001/exe/a.out -importcfg $WORK/b001/importcfg.link -buildmode=exe -buildid=TRkhr1DOncFvwJGBuBqF/z7-eiWaVSWcDdOOZo7Ws/ivVyZcwbGRh1UXmmIxq7/TRkhr1DOncFvwJGBuBqF -extld=gcc $WORK/b001/_pkg_.a
/usr/local/go/pkg/tool/linux_amd64/buildid -w $WORK/b001/exe/a.out # internal
mkdir -p /home/arthur/go/bin/
cp $WORK/b001/exe/a.out /home/arthur/go/bin/bbgo
rm -r $WORK/b001/
arthur@crypto:~$ ls -lh /home/arthur/go/bin/bbgo
-rwxrwxr-x 1 arthur arthur 34M Dec 21 17:05 /home/arthur/go/bin/bbgoarthur@crypto:~$ /home/arthur/go/bin/bbgo version1.x-devarthur@crypto:~$ cp /home/arthur/go/bin/bbgo bbgo/bbgocp: cannot create regular file 'bbgo/bbgo': Text file busyarthur@crypto:~$ killall node
arthur@crypto:~$ killall bbgo
bbgo: no process found
arthur@crypto:~$ cp /home/arthur/go/bin/bbgo bbgo/bbgo
arthur@crypto:~$ timeout --kill-after=630s 600s /home/arthur/bbgo/bbgo backtest --exchange binance --dotenv /tmp/bbgo-env/bbgo-env-KAVAUSDT.yaml --config /tmp/bbgo-config/37236810dc639fd9269d97f98ae45bc3.yaml --base-asset-baseline --output /tmp/bbgo-cache/37236810dc639fd9269d97f98ae45bc3/ 2>&1
panic: <APIError> code=0, msg=

goroutine 1 [running]:
github.com/c9s/bbgo/pkg/exchange/binance.New({0xc0001940b0, 0x40}, {0xc0000e81f3, 0x40})
        /home/arthur/go/pkg/mod/github.com/c9s/bbgo@v1.20.1-0.20211221165413-8b93aeeeb277/pkg/exchange/binance/exchange.go:69 +0x5bb
github.com/c9s/bbgo/pkg/cmd/cmdutil.NewExchangeStandard({0xc00090d458, 0x11dfaeb}, {0xc0001940b0, 0xc00068ad80}, {0xc0000e81f3, 0xc000915100}, {0x0, 0xc00068a510}, {0x0, 0x0})
        /home/arthur/go/pkg/mod/github.com/c9s/bbgo@v1.20.1-0.20211221165413-8b93aeeeb277/pkg/cmd/cmdutil/exchange.go:22 +0x12d
github.com/c9s/bbgo/pkg/cmd/cmdutil.NewExchangeWithEnvVarPrefix({0xc00090d458, 0x7}, {0xc00090d468, 0x0})
        /home/arthur/go/pkg/mod/github.com/c9s/bbgo@v1.20.1-0.20211221165413-8b93aeeeb277/pkg/cmd/cmdutil/exchange.go:51 +0x213
github.com/c9s/bbgo/pkg/bbgo.InitExchangeSession({0x0, 0x36}, 0xc0004e7a00)
        /home/arthur/go/pkg/mod/github.com/c9s/bbgo@v1.20.1-0.20211221165413-8b93aeeeb277/pkg/bbgo/session.go:696 +0xad
github.com/c9s/bbgo/pkg/cmd.glob..func2(0x1b628c0, {0x11d3868, 0x9, 0x9})
        /home/arthur/go/pkg/mod/github.com/c9s/bbgo@v1.20.1-0.20211221165413-8b93aeeeb277/pkg/cmd/backtest.go:116 +0x745
github.com/spf13/cobra.(*Command).execute(0x1b628c0, {0xc0006ba360, 0x9, 0x9})
        /home/arthur/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:850 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0x1b635e0)
        /home/arthur/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x3ad
github.com/spf13/cobra.(*Command).Execute(...)
        /home/arthur/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895
github.com/c9s/bbgo/pkg/cmd.Execute()
        /home/arthur/go/pkg/mod/github.com/c9s/bbgo@v1.20.1-0.20211221165413-8b93aeeeb277/pkg/cmd/root.go:178 +0x438
main.main()
        /home/arthur/go/pkg/mod/github.com/c9s/bbgo@v1.20.1-0.20211221165413-8b93aeeeb277/cmd/bbgo/main.go:8 +0x17


```

**Comment by c9s** [comment]
the second error seems unrelated, but the first error should be fixed in commit `8b93aeeeb277b4fd8804f41bd5509cc3a06d982c` and the second is fixed in `bace22547017a8701cb7aa0c75916ed0bec6762a`
